12b5 + Ruby documentation (#4650)
circleci/circleci-docs: jekyll/_cci2/testing-ios.md

**Note:** Installing Gems with the system Ruby is not advised due to the restrictive permissions enforced on the system directories. As a general rule, we advise using one of the alternative Rubies provided by Chruby for all jobs.

### Images using Xcode 12 and later

As a result of the macOS system Ruby (2.6.3) becoming increasingly incompatible with various gems (especially those which require native extensions), Xcode 12 and later iamges default to Ruby 2.7 via `chruby`. 

> This allows for greater compatibility and reliability out of the box with gems moving forward. For the vast majority of users using Ruby 2.7 will feel transparent as common gems, such as Fastlane, run without any issues.

Reviewer: ```suggestion
Defaulting to Ruby 2.7 allows for greater compatibility and reliability with gems moving forward. Common gems, such as Fastlane, run without any issues in Ruby 2.7.
```

> To switch to another Ruby version see [our chruby documentation](#images-using-macos-1015-xcode-112-and-later), or to revert back to the system Ruby, add the following to the beginning of your job:

Reviewer: ```suggestion
To switch to another Ruby version, see [our chruby documentation](#images-using-macos-1015-xcode-112-and-later). To revert back to the system Ruby, add the following to the beginning of your job:
```
